service/module.go

```diff
@@ -119,16 +129,26 @@ func newModuleWrapper(
 			return nil, err
 		}
 	}
-	if moduleOptions.name == "" {
-		moduleOptions.name = moduleProvider.DefaultName()
+	var name string
+	var roles []string
+	if moduleOptions.ModuleName == "" {
+		moduleOptions.ModuleName = moduleProvider.DefaultName()
 	}
-	scopedHost, err := newScopedHost(
-		host,
-		moduleOptions.name,
-		moduleOptions.roles...,
-	)
-	if err != nil {
-		return nil, err
+	if moduleOptions.ServiceName != "" {
```

Comment:
It's not the same though. There is an else here too to set to host's values. Optionally, I can move this to the function implementation in ScopedHost but couldn't decide which was better and stuck with this :)